File: Codigos estudiantes por lenguaje/PY/Daniel Cangrejo/Ordenamientos/Comb_Sort.py
from sys import stdin,stdout
rl = stdin.readline
wr = stdout.write


def getNextGap(gap):
    gap = (gap * 10) // 13
    if gap < 1:
        return 1
    return gap


def combSort(arr, n):

    gap = n
    swapped = True
    
    while gap != 1 or swapped:
        gap = getNextGap(gap)
        swapped = False
        for i in range(n-gap):
            if arr[i] > arr[i+gap]:
                arr[i], arr[i+gap] = arr[i+gap], arr[i]
                swapped = True

    
arr = list(map(int, rl().split()))
combSort(arr, len(arr))
print(*arr)

File: Codigos estudiantes por lenguaje/C++/Daniel Cangrejo/Ordenamientos/Comb_Sort.cpp
include<bits/stdc++.h>
#include<cstdlib>
using namespace std;
int getNextGap(int gap){
    gap=(gap*10)/13;
    if(gap<1){
        return 1;
    }
    return gap;
}
void sortArr(int arr[],int n){
    int gap=n;
    bool swapped=true;
    while(gap!=1 || swapped == true){
        gap=getNextGap(gap);
        swapped=false;
        for(int i=0;i<n-gap;i++){
            if(arr[i]>arr[i+gap]){
                swap(arr[i],arr[i+gap]);
            swapped=true;
            }
        }
    }
}
void printArr(int arr[],int n){
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}
int main() {
    int arr[]={8,4,1,56,3,-44,23,-6,28,0};
    int n=sizeof(arr)/sizeof(arr[0]);
    sortArr(arr,n);
    cout<<"arreglo ordenado"<<endl;
    printArr(arr,n);
}
